### ISSUE TITLE
 Unexpected ==. For assignment, use =.

### ISSUE THREAD

**Issue by Earnestly** [koalaman/shellcheck#1087]
* shellcheck version: 0.4.7
* rule id: SC1097
* [x] reproducible on shellcheck.net

```sh
#!/bin/sh --
while IFS== read -r a; do
    :
done
```

When using `shellcheck --format=gcc foo` I get the following output:

```
foo:2:10: error: Unexpected ==. For assignment, use =. For comparison, use [/[[. [SC1097]
```

A potential workaround is to quote the assignment of the `=` but that shouldn't be necessary.

**Comment by koalaman** [comment]
I agree that it shouldn't be necessary to modify perfectly valid code to please tooling. 

Most of the time though, this would correctly trigger for some newbie's `while var==value` or `if ($var == value)`. Please consider helping out such users by making your intent more explicit with `IFS="="`.

You can also ignore this warning since the snippet is still being correctly parsed.

**Comment by Earnestly** [comment]
While I don't mind the workaround provided here as I happen to agree with the opinion; Doesn't this issue still exist as a semantic failure of the tool?

Perhaps it might be useful if you used or added a `wontfix` label since I assume that's the position here.
